src/lib/ngsiNotify/Notifier.h

```diff
@@ -25,33 +25,37 @@
 *
 * Author: Fermin Galan
 */
-
 #include <map>
 #include <pthread.h>
 
+#include "common/NotificationFormat.h"
 #include "ngsi9/NotifyContextAvailabilityRequest.h"
 #include "ngsi10/NotifyContextRequest.h"
+#include "ngsiNotify/ThreadData.h"
 
-#include "ThreadData.h"
 
-class Notifier {
 
+/* ****************************************************************************
+*
+* Notifier -
+*/
+class Notifier
+{
 public:
-   
-    virtual ~Notifier(void);
-
-    virtual void sendNotifyContextRequest(NotifyContextRequest* ncr,
-                                          const std::string&    url,
-                                          const std::string&    tenant,
-                                          const std::string&    xauthToken,
-                                          const std::string&    fiwareCorrelator,
-                                          Format                format   = JSON);
-
-    virtual void sendNotifyContextAvailabilityRequest(NotifyContextAvailabilityRequest* ncr,
-                                                      const std::string&                url,
-                                                      const std::string&                tenant,
-                                                      const std::string&                fiwareCorrelator,
-                                                      Format                            format           = JSON);
+  virtual ~Notifier(void);
+
+  virtual void sendNotifyContextRequest(NotifyContextRequest* ncr,
+                                        const std::string&    url,
+                                        const std::string&    tenant,
+                                        const std::string&    xauthToken,
+                                        const std::string&    fiwareCorrelator,
+                                        NotificationFormat    notifyFormat = NGSI_V2_NORMALIZED);
```

Comment:
If default value is tricky, then let's remove default values at all.

These method are called from a couple or places or so, so it shouldn't be difficult ot adjust... at least in the "actual" code (not sure about unit test, that may involve a number between 0 or thousand changes, based in previous experiences :). If unit test are a problem, then, let's pospone the un-defaulting to a next PR.